File: vanpaayaar-backend/src/main/java/com/tribal/service/SellerService.java
package com.tribal.service;

import com.tribal.model.Seller;
import com.tribal.model.Product;
import com.tribal.model.Order;

import java.util.List;
import java.util.Map;

public interface SellerService {

    Seller getSellerById(Long sellerId);
    Seller updateProfile(Long sellerId, Seller updatedSeller);
    void deleteSeller(Long sellerId);

    Product addProduct(Long sellerId, Product product);
    List<Product> getProductsBySeller(Long sellerId);
    Product updateProduct(Long productId, Product product);
    void deleteProduct(Long productId);

    Seller registerSeller(Seller seller);
    Seller updateApprovalStatus(Long sellerId, String status);

    List<Order> getOrdersBySeller(Long sellerId);
    Order updateOrderStatus(Long orderId, String status);

    Map<String, Object> getSellerDashboard(Long sellerId);
}

File: vanvyapaar-frontend/src/services/sellerService.ts
import api from '../lib/api'
import { Product, Order } from '../types'

export interface SellerStats {
  totalProducts: number
  pendingOrders: number
  monthlySales: number
  profileViews: number
  totalSales?: number
  totalOrders?: number
}

export const sellerService = {
  // Dashboard stats - Fixed to match backend endpoint
  getStats: async (sellerId: number): Promise<SellerStats> => {
    const response = await api.get(`/seller/${sellerId}/dashboard`)
    const data = response.data
    // Map backend response to frontend interface
    return {
      totalProducts: data.totalProducts || 0,
      pendingOrders: data.pendingOrders || 0,
      monthlySales: data.monthlySales || 0, // Monthly sales (last 30 days) - matches analytics "month" period
      profileViews: 0, // Not implemented in backend yet
      totalSales: data.totalSales || 0, // All-time total sales
      totalOrders: data.totalOrders || 0
    }
  },

  // Product operations
  getProducts: async (sellerId: number): Promise<Product[]> => {
    const response = await api.get(`/seller/${sellerId}/products`)
    return response.data
  },

  addProduct: async (sellerId: number, product: Partial<Product>): Promise<Product> => {
    const response = await api.post(`/seller/${sellerId}/products`, product)
    return response.data
  },

  updateProduct: async (productId: number, product: Partial<Product>): Promise<Product> => {
    const response = await api.put(`/seller/products/${productId}`, product)
    return response.data
  },

  deleteProduct: async (productId: number): Promise<void> => {
    await api.delete(`/seller/products/${productId}`)
  },

  // Order operations
  getOrders: async (sellerId: number): Promise<Order[]> => {
    const response = await api.get(`/seller/${sellerId}/orders`)
    return response.data
  },

  updateOrderStatus: async (orderId: number, status: string): Promise<Order> => {
    const response = await api.put(`/seller/orders/${orderId}/status?status=${status}`)
    return response.data
  },

  // Analytics
  getAnalytics: async (sellerId: number, period: string = 'month'): Promise<any> => {
    const response = await api.get(`/seller/${sellerId}/analytics?period=${period}`)
    return response.data
  }
}

export default sellerService
